library/pkparse.c

```diff
@@ -970,6 +968,8 @@ static int pk_parse_key_pkcs8_encrypted_der(
      *  EncryptedData ::= OCTET STRING
      *
      *  The EncryptedData OCTET STRING is a PKCS#8 PrivateKeyInfo
+     *
+     * To save space, the decryption happens in-place on the given key buffer.
```

Comment:
An error in the middle of the function will leave partial output in the output buffer. I think we should zero out `key` before returning if an error occurs, to avoid the risk that the caller will misuse an invalid result if they aren't checking error codes properly.

Reply:
@gilles-peskine-arm Currently the code puts the responsibility of zeroizing the buffer on the caller, see https://github.com/hanno-arm/mbedtls/blob/cc5662811729486ba5f063e97e312f664bb7377c/library/pkparse.c#L1143 and https://github.com/hanno-arm/mbedtls/blob/cc5662811729486ba5f063e97e312f664bb7377c/library/pkparse.c#L1166. I have added a commit explicitly documenting that - are you ok with it?

Reply:
Ok, because it's a static function.